Bearded.Utilities/Threading/BackgroundActionQueue.cs

```diff
@@ -129,7 +129,7 @@ public T RunAndReturn<T>(Func<T> action)
         {
             var ret = default(T);
             RunAndAwait(() => ret = action());
-            return ret;
+            return ret!;
```

Comment:
The implementation of `RunAndAwait` ensures that the return value will be set. It may of course be set to null, if that is what the action returns, but that is entirely up to the caller.